File: projects/SonarQ/source_bug/test_vuln.js
const express = require('express');
const app = express();
const mysql = require('mysql');

// Hard-coded credentials
const API_KEY = 'sk-1234567890abcdef';
const DB_PASSWORD = 'admin123';

// SQL Injection vulnerability
app.get('/user', (req, res) => {
    const userId = req.query.id;
    const query = `SELECT * FROM users WHERE id = ${userId}`; // SQL injection
    res.send(query);
});

// Command injection
app.get('/ping', (req, res) => {
    const host = req.query.host;
    const { exec } = require('child_process');
    exec(`ping ${host}`, (error, stdout, stderr) => { // Command injection
        res.send(stdout);
    });
});

// XSS vulnerability
app.get('/search', (req, res) => {
    const query = req.query.q;
    res.send(`<h1>Search results for: ${query}</h1>`); // XSS
});

app.listen(3000);

File: projects/SonarQ/source_bug/test_vuln.py
import os
import subprocess
from flask import Flask, request

app = Flask(__name__)

# SQL Injection vulnerability
@app.route('/user')
def get_user():
    user_id = request.args.get('id')
    query = f"SELECT * FROM users WHERE id = {user_id}"  # SQL injection
    return query

# Command injection vulnerability
@app.route('/ping')
def ping():
    host = request.args.get('host')
    result = os.system(f"ping {host}")  # Command injection
    return str(result)

# Hard-coded secret
API_KEY = "sk-1234567890abcdef"  # Hard-coded API key
DATABASE_PASSWORD = "admin123"  # Hard-coded password

if __name__ == '__main__':
    app.run(debug=True)  # Debug mode in production